CRAM-no-header-md5sum/md5sum/CRAM_tool_samtools.cwl

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env cwl-runner
+
+class: CommandLineTool
+id: CRAM_samtools
+label: samtools tool
+cwlVersion: v1.0
+
+$namespaces:
+  dct: http://purl.org/dc/terms/
+  foaf: http://xmlns.com/foaf/0.1/
+
+dct:creator:
+  '@id':  https://orcid.org/0000-0001-5173-4627
+  foaf:name: Walter Shands
+  foaf:mbox: jshands@ucsc.edu
+
+requirements:
+- class: DockerRequirement
+  dockerPull: us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.3.2-1510681135
+- class: InlineJavascriptRequirement
+
+hints:
+- class: ResourceRequirement
+  # The command uses primarily one core but significant disk space.
+  coresMin: 1
+  ramMin: 1024
+  outdirMin: 100000
+
+requirements:
+  InitialWorkDirRequirement:
+    listing:
+      - $(inputs.input_file)
+      - $(inputs.reference_file)
+      - $(inputs.reference_index_file)
+
+inputs:
+  reference_file:
+    type: File
+    inputBinding:
+      position: 1
+      prefix: -T
+    doc: The genome reference file for the BAM or CRAM file.
+
+  reference_index_file:
+    type: File
+    inputBinding:
+      position: 2
+      prefix: -t
```

Comment:
Slightly confusing that this prefix and other prefix only differ by case. Could you pick more meaningful values?

Reply:
Or is this because this is what SAM tools is expecting?

Reply:
These are samtools parameters http://www.htslib.org/doc/samtools-1.0.html 
-T A FASTA format reference FILE, optionally compressed by bgzip and ideally indexed by samtools faidx. If an index is not present, one will be generated for you.
-t Genome reference index file. If you run: `samtools faidx <ref.fa>', the resulting index file <ref.fa>.fai can be used as this FILE.